datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -1278,6 +1279,40 @@ pub(crate) fn from_substrait_literal(lit: &Literal) -> Result<ScalarValue> {
                 s,
             )
         }
+        Some(LiteralType::List(l)) => {
+            let elements = l
+                .values
+                .iter()
+                .map(|el| from_substrait_literal(el))
+                .collect::<Result<Vec<_>>>()?;
+            let element_type = elements[0].data_type();
```

Comment:
Should we check if elements are empty and report an error? The literal input might come from systems other than DataFusion, and they might not be properly implemented.

Reply:
Yup, done: cdc525cef141e87c74521c146c54e69863b35084